### PR TITLE
Enable ceres & suitesparse for Windows.

### PR DESCRIPTION
**Pull request by seanyen**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by seanyen** [comment]
@conda-forge-admin, please rerender

**Comment by seanyen** [comment]
hmm, it seems the [`suitesparse-feedstock`](https://github.com/conda-forge/suitesparse-feedstock) needs a new build?

```
Package openblas conflicts for:
suitesparse=5.1 -> openblas[version='0.2.20|0.2.20.*|>=0.3.3,<0.3.4.0a0']
openblas=0.3.6
suitesparse=5.1 -> blas==1.1=openblas -> openblas
blas==1.1=openblas -> openblas
```

**Comment by seanyen** [comment]
Waiting for this. https://github.com/conda-forge/suitesparse-feedstock/pull/65

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,8 +10,10 @@ source:
   sha256: 4744005fc3b902fed886ea418df70690caa8e2ff6b5a90f3dd88a3d291ef8e8e
 
 build:
-  number: 7
+  number: 8
   skip: true  # [win and vc<14]
+  features:  # [win]
+    - blas_{{ variant }}  # [win]
```
Comment (jschueller) [inline recipe/meta.yaml:16]: the blas variant feature is deprecated, please remove this

File: recipe/meta.yaml [hunk 2]
```diff
@@ -23,17 +26,21 @@ requirements:
...
     - gflags
-    - libblas
-    - libcblas
-    - liblapack
-    - suitesparse  # [unix]
+    - libblas  # [unix]
+    - libcblas  # [unix]
+    - liblapack  # [unix]
+    - blas 1.1 {{ variant }}  # [win]
+    - openblas  # [win]
```
Comment (jschueller) [inline recipe/meta.yaml:33]: here too, remove blas variant & openblas

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {% set version = "1.14.0" %}
+{% set variant = "openblas" %}
```
Comment (jschueller) [inline recipe/meta.yaml:2]: you can also drop this

File: recipe/meta.yaml [hunk 4]
```diff
@@ -23,17 +24,19 @@ requirements:
...
+    - libblas  # [unix]
+    - libcblas  # [unix]
+    - liblapack  # [unix]
+    - blas 1.1  # [win]
+    - suitesparse
     - tbb
     - eigen
 
   run:
+    - blas 1.1  # [win]
```
Comment (jschueller) [inline recipe/meta.yaml:36]: and this

**Comment by seanyen** [comment]
@conda-forge-admin, please rerender

**Comment by seanyen** [comment]
@jschueller Thanks for those tips! It seems now the `conda` is happy about the dependencies.

**Comment by seanyen** [comment]
@jschueller Btw, I am adding myself as a co-maintainer. Hope you don't mind. :)

**Comment by seanyen** [comment]
@jschueller Sorry to spam again. I think now the change is ready for merge. I am planning to add https://github.com/cartographer-project/cartographer feedstock and which depends on this change. Thanks!

File: recipe/meta.yaml [hunk 5]
```diff
@@ -23,17 +23,18 @@ requirements:
...
     - glog
     - gflags
-    - libblas
-    - libcblas
-    - liblapack
-    - suitesparse  # [unix]
+    - libblas  # [unix]
+    - libcblas  # [unix]
+    - liblapack  # [unix]
+    - blas 1.1  # [win]
```
Comment (jschueller) [inline recipe/meta.yaml:29]: blas 1.1 should not be here either

**Comment by seanyen** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 6]
```diff
@@ -26,7 +26,7 @@ requirements:
     - libblas  # [unix]
     - libcblas  # [unix]
     - liblapack  # [unix]
-    - blas 1.1  # [win]
+    - blas  # [win]
```
Comment (jschueller) [inline recipe/meta.yaml:29]: there is no blas package, only libblas

Comment (seanyen) [reply recipe/meta.yaml:29]: Sorry I don’t follow. What does that mean? Btw, for the original my change, I was following https://conda-forge.readthedocs.io/en/latest/blas.html and the example in suitesparse-feedstock. I am pretty much just “monkey see monkey do”.

Comment (jschueller) [reply recipe/meta.yaml:29]: You dont need this since ceres doesnt care about the implementation: libblas is enough.

Comment (seanyen) [reply recipe/meta.yaml]: Ah, I see. Just a thought, maybe for easier maintenance, we can remove the `libblas`, `libclas`, and 'liblapack` all together? They are deps from `suitesparse` anyway.

**Comment by seanyen** [comment]
@conda-forge-admin, please rerender

**Comment by seanyen** [comment]
@jschueller Let me know any additional changes to make. Hope we can get this enabled for Windows build. Thanks.

**Comment by seanyen** [comment]
@conda-forge-admin, please rerender

**Comment by seanyen** [comment]
@jschueller The new `suitesparse` is merged and now on the `conda-forge`. And this is ready for review and merge. Thanks!

**Comment by jschueller** [comment]
you build with suitesparse 5.4 but for runtime 5.1 is still pulled, this seems inconsistent to me.
maybe you'd want to pin suitesparse to 5.4 for the run requirements too, unless the pinning update gets merged first:
https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/564

**Comment by seanyen** [comment]
@jschueller Thanks for pointing out the version issue. I added the `run_exports` to the `suitesparse`, and by that way it should pull the version with it built against. https://github.com/conda-forge/suitesparse-feedstock/pull/66

**Comment by seanyen** [comment]
@conda-forge-admin, please rerender

**Comment by seanyen** [comment]
@jschueller Finally the `suitesparse` pinning went through! And the CI looks green and consumes the latest `5.4` `suitesparse` for Windows.

**Pull request merged**